xml_converter/src/string_helper.cpp

```diff
@@ -287,3 +287,10 @@ bool has_suffix(std::string const& fullString, std::string const& ending) {
         return false;
     }
 }
+
+void ensure_trailing_slash(std::string* directory_path) {
```

Comment:
As discussed offline, this is only used when you want to join two string paths together. This function should just do that instead needing to remove all the const references